OctaneVSPlugin/ConnectionSettings.cs

```diff
@@ -0,0 +1,246 @@
+﻿/*!
+* (c) 2016-2018 EntIT Software LLC, a Micro Focus company
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using MicroFocus.Adm.Octane.Api.Core.Connector;
+using MicroFocus.Adm.Octane.Api.Core.Connector.Authentication;
+using MicroFocus.Adm.Octane.VisualStudio.Common;
+using MicroFocus.Adm.Octane.VisualStudio.View;
+using MicroFocus.Adm.Octane.VisualStudio.ViewModel;
+using Microsoft.VisualStudio.Shell;
+using System;
+using System.ComponentModel;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using static System.Threading.Tasks.Task;
+
+namespace MicroFocus.Adm.Octane.VisualStudio
+{
+	/// <summary>
+	/// Options page represent the settings the user can set for ALM Octane plugin.
+	/// This class is presented to the user as a page in Visual Studio options dialog.
+	/// </summary>
+	[Guid("1D9ECCF3-5D2F-4112-9B25-264596873DC9")]
+	internal class ConnectionSettings : UIElementDialogPage
+	{
+
+		public const string ConnectionSuccessful = "Connection successful!";
+		private string url = string.Empty;
+		private int ssid = 1001;
+		private int wsid = 1002;
+		private string user = string.Empty;
+		private string password = string.Empty;
+		private bool credentialLogin = true;
+		private bool ssologin = false;
+		private ConnectionSettingsView page;
+
+		protected override void OnApply(PageApplyEventArgs e)
+		{
+
+			page.SetInfoLabelText("");
+			SaveSettingsToStorage();
+			var result = Run(async () => { return await TestConnection(); }).Result;
+
+			if (!result.Equals(ConnectionSuccessful))
+			{
+				e.ApplyBehavior = ApplyKind.CancelNoNavigate;
+			}
+
+			else
+			{
+				// reset and thus require a new octane service obj
+				Run(async () => { return await OctaneServices.Reset(); }).Wait();
+
+				// create a new service object
+				OctaneServices.Create(OctaneConfiguration.Url,
+						  OctaneConfiguration.SharedSpaceId,
+						  OctaneConfiguration.WorkSpaceId);
+
+                
+                // close all opened details windows so that we don't have details windows
+                // for entities from different workspaces
+                PluginWindowManager.CloseAllDetailsWindows();
```

Comment:
need to find a way to fix the messed up indentation in vs

Reply:
github is messing with the indentation
looks ok to me
![image](https://user-images.githubusercontent.com/30106237/51179091-436ba080-18cd-11e9-9231-9ef326b00c86.png)